File: code-graph-indexer/evals/fixtures/sample.ts
/** Small TypeScript fixture for indexer smoke tests. */
import { createLogger } from "./log";

const logger = createLogger("sample");

export class Greeter {
    /** Greet someone by name. */
    greet(name: string): string {
        return formatMessage(name);
    }
}

export function formatMessage(name: string): string {
    try {
        return `Hello, ${name}!`;
    } catch (e) {
        logger.error("format failed", e);
        throw e;
    }
}

export function silentDivide(a: number, b: number): number | null {
    try {
        return a / b;
    } catch (e) {
        return null;
    }
}

File: code-graph-indexer/evals/fixtures/Sample.java
package evals.fixtures;

import java.util.logging.Logger;

/** Small Java fixture for indexer smoke tests. */
public class Sample {
    private static final Logger logger = Logger.getLogger(Sample.class.getName());

    /** Public entry point; calls the private helper. */
    public String greet(String name) {
        return formatMessage(name);
    }

    private String formatMessage(String name) {
        try {
            return "Hello, " + name + "!";
        } catch (RuntimeException e) {
            logger.warning("format failed: " + e);
            throw e;
        }
    }

    public Integer silentDivide(int a, int b) {
        try {
            return a / b;
        } catch (ArithmeticException e) {
            return null;
        }
    }
}
